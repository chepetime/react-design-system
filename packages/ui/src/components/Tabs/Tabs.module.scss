@use "./../../styles" as styles;

:root {
  --tabs-background-color: #{styles.$pink-500};
}

:global .dark {
  --tabs-background-color: #{styles.$indigo-500};
}

.Tabs {
  @include styles.styles-reset;
  @include styles.text-wrapping;

  background: var(--tabs-background-color);

  font-size: 1rem;

  @include styles.respond-to("md") {
    font-size: 1.5rem;
  }
  @include styles.respond-to("lg") {
    font-size: 2rem;
  }
  @include styles.feature-to("hover") {
    &:hover {
      background-color: pink;
    }
  }
}
