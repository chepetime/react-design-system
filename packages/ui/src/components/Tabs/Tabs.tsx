import React from "react";
import classnames from "classnames";
import css from "./Tabs.module.scss";

export interface TabsProps extends React.ComponentPropsWithRef<"div"> {
  children?: React.ReactNode;
}

export const Tabs = React.forwardRef<HTMLDivElement, TabsProps>(
  (props: TabsProps, ref: React.Ref<HTMLDivElement>) => {
    return (
      <div className={classnames(css.Tabs)} ref={ref}>
        <p>Tabs component working!</p>
        <div>{props.children}</div>
      </div>
    );
  }
)

Tabs.displayName = "Tabs";
