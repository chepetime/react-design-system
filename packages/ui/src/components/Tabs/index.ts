export { Tabs } from "./Tabs";
export type { TabsProps } from "./Tabs";