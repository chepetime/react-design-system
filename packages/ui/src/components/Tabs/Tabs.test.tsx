import React from "react";import { render } from "@testing-library/react";
import { Tabs } from "./Tabs";

describe("Tabs", () => {
  test("renders the Tabs component", () => {
    render(<Tabs />);
  });
});
