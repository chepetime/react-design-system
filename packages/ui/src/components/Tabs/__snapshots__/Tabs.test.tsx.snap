// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs matches snapshot 1`] = `
<div>
  <div
    class="Tabs"
  >
    <p>
      Tabs component working!
    </p>
    <div />
  </div>
</div>
`;
